packages/labextension/pyproject.toml

```diff
@@ -0,0 +1,64 @@
+[build-system]
+requires = ["hatchling>=1.5.0", "jupyterlab>=4.0.0,<5", "hatch-nodejs-version"]
+build-backend = "hatchling.build"
+
+[project]
+name = "jupyterlab_jupytext"
+readme = "README.md"
+license = { file = "LICENSE" }
+requires-python = ">=3.8"
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "License :: OSI Approved :: MIT License",
+    "Environment :: Console",
+    "Framework :: Jupyter",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Science/Research",
+    "Topic :: Text Processing :: Markup",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+]
+dependencies = [
+    "jupyterlab>=4.0.0,<5",
```

Comment:
Do not upper pin dependencies. Instead add an `experimental` CI matrix job that runs for future dependencies like `jupyterlab ~= 5.0`.

Also simplify the lowerbound to `4.0` or `4`. Bugfix information is not useful, unless there is a specific bug that needs to be avoided.

Reply:
In principle, but keep in mind that this makes it harder and slower to maintain. When Sphinx 7 came out just a few months after Sphinx 6 came out, there were a bunch of dependency hell where so many packages had circular dependencies in tests and main dependency making the whole updating process unnecessarily complicated. Same with this package.

So it's not only about the users installing the packages, but especially any dependent package, particularly if the upper-bound is in `dependencies` rathet than `optional-dpendencies`. It is best to keep upper-bound unbounded, but instead have the CI check for compatibility issues.

Reply:
Fair enough, I see your point. I will remove the upper bound. Cheers!